java/src/test/java/org/apache/arrow/gandiva/evaluator/MicroBenchmark.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2017-2018 Dremio Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.gandiva.evaluator;
+
+import com.google.common.collect.Lists;
+import org.apache.arrow.gandiva.exceptions.GandivaException;
+import org.apache.arrow.gandiva.expression.ExpressionTree;
+import org.apache.arrow.gandiva.expression.TreeBuilder;
+import org.apache.arrow.gandiva.expression.TreeNode;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MicroBenchmark extends BaseNativeEvaluatorTest {
+
+  @Test
+  public void testAdd3() throws GandivaException, Exception {
+    Field x = Field.nullable("x", int32);
+    Field N2x = Field.nullable("N2x", int32);
+    Field N3x = Field.nullable("N3x", int32);
+
+    List<TreeNode> args = new ArrayList<TreeNode>();
+
+    // x + N2x + N3x
+    TreeNode add1 = TreeBuilder.makeFunction("add", Lists.newArrayList(TreeBuilder.makeField(x), TreeBuilder.makeField(N2x)), int32);
+    TreeNode add = TreeBuilder.makeFunction("add", Lists.newArrayList(add1, TreeBuilder.makeField(N3x)), int32);
+    ExpressionTree expr = TreeBuilder.makeExpression(add, x);
+
+    List<Field> cols = Lists.newArrayList(x, N2x, N3x);
+    Schema schema = new Schema(cols);
+
+    long timeTaken = timedEvaluate(new Int32DataAndVectorGenerator(allocator),
+            schema,
+            Lists.newArrayList(expr),
+            100 * MILLION, 16 * THOUSAND,
+            4);
+    System.out.println("Time taken for evaluating 100m records of add3 is " + timeTaken + "ms");
+  }
+
+  // TODO: Using int64 in testIf causes the test to fail due to lack of memory
```

Comment:
even with a batch size of 16K, the men used should be in the range of a few MBs. Memory leak ??